test/style/custom_vector_source.test.cpp

```diff
@@ -0,0 +1,53 @@
+#include <mbgl/test/util.hpp>
+#include <mbgl/style/sources/geojson_source.hpp>
+
+#include <mbgl/util/geo.hpp>
+
+#include <mbgl/style/sources/custom_vector_source.hpp>
+
+using namespace mbgl;
+
+
+TEST(CustomVectorSource, EmptyData) {
+    mbgl::style::GeoJSONOptions options;
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wunused-parameter"
+    const auto callback = ^void(uint8_t z, uint32_t x, uint32_t y) {};
```

Comment:
Can we instead use C++11 lambdas rather than Apple-Clang specific syntax? You can also remove the z/x/y names to avoid the pragmas. An even better approach would be to pass a null pointer as the callback and check whether `callback` is invokable before calling it.